sympy/core/power.py

```diff
@@ -1622,7 +1622,7 @@ def _eval_nseries(self, x, n, logx, cdir=0):
         g = (b/f - S.One).cancel()
         maxpow = n - m*e
 
-        if maxpow < S.Zero:
+        if maxpow.is_negative:
```

Comment:
This change is with ref to: https://github.com/sympy/sympy/issues/20897

Reply:
Did you encounter an example where this would raise?

Reply:
I had encountered the case with the condition on `Mul._eval_nseries` changed, but not with the current condition. I think it should be reproducible, will try and get back to you.

Reply:
It looks like `n`, `m` and `e` would be real numbers. (Otherwise there is probably a bug somewhere in the leading term code.)

Reply:
They are indeed real, but zero checking isn't always possible as Oscar had pointed out in the issue.
```py
In [1]: a = (5*log(5)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) + 1)

In [2]: a.evalf()
Out[2]: 0.e-125

In [3]: if a < S.Zero:
   ...:     print(".")
   ...: 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-b716c5be69d4> in <module>
----> 1 if a < S.Zero:
      2     print(".")

~/sympy/sympy/core/relational.py in __bool__(self)
    396 
    397     def __bool__(self):
--> 398         raise TypeError("cannot determine truth value of Relational")
    399 
    400     def _eval_as_set(self):

TypeError: cannot determine truth value of Relational
```

Reply:
While not the simplest form, this example raises an error on the current master -
```py
In [1]: w, p = symbols('w p')

In [2]: logx = -p*log(x) + 6*p*log(3*p + 1) - 5*p*log(5*p + 2) - 6*p*log(6) + 6*p*log(2)

In [3]: x = w

In [4]: a = 1/(180*p**2*w**(5*log(5)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)))*sqrt(3*p + 1)*exp(5*p*log(5)*log(p)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*lo
   ...: g(3)) - 30*p*log(5)*log(3*p + 1)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) + 25*p*log(5)*log(5*p + 2)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3))
   ...: + 30*p*log(5)*log(6)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) - 30*p*log(2)*log(5)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) + 5*p*log(5)) + 12
   ...: 0*w**(5*log(5)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)))*p*sqrt(3*p + 1)*exp(5*p*log(5)*log(p)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) - 30*p
   ...: *log(5)*log(3*p + 1)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) + 25*p*log(5)*log(5*p + 2)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) + 30*p*log(5
   ...: )*log(6)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) - 30*p*log(2)*log(5)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) + 5*p*log(5)) + 20*w**(5*log(5
   ...: )/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)))*sqrt(3*p + 1)*exp(5*p*log(5)*log(p)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) - 30*p*log(5)*log(3*p
   ...:  + 1)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) + 25*p*log(5)*log(5*p + 2)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) + 30*p*log(5)*log(6)/(-6*lo
   ...: g(6) - 5*log(5) + 6*log(2) + 6*log(3)) - 30*p*log(2)*log(5)/(-6*log(6) - 5*log(5) + 6*log(2) + 6*log(3)) + 5*p*log(5)))

In [5]: a.nseries(x, 1, logx)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-eb9556bec087> in <module>
----> 1 a.nseries(x, 1, logx)

~/sympy/sympy/core/expr.py in nseries(self, x, x0, n, dir, logx, cdir)
   3335             return self
   3336         if x is None or x0 or dir != '+':  # {see XPOS above} or (x.is_positive == x.is_negative == None):
-> 3337             return self.series(x, x0, n, dir, cdir=cdir)
   3338         else:
   3339             return self._eval_nseries(x, n=n, logx=logx, cdir=cdir)

~/sympy/sympy/core/expr.py in series(self, x, x0, n, dir, logx, cdir)
   2971                 rep2 = x
   2972                 rep2b = -x0
-> 2973             s = self.subs(x, rep).series(x, x0=0, n=n, dir='+', logx=logx, cdir=cdir)
   2974             if n is None:  # lseries...
   2975                 return (si.subs(x, rep2 + rep2b) for si in s)

~/sympy/sympy/core/expr.py in series(self, x, x0, n, dir, logx, cdir)
   2981             # replace x with an x that has a positive assumption
   2982             xpos = Dummy('x', positive=True, finite=True)
-> 2983             rv = self.subs(x, xpos).series(xpos, x0, n, dir, logx=logx, cdir=cdir)
   2984             if n is None:
   2985                 return (s.subs(xpos, x) for s in rv)

~/sympy/sympy/core/expr.py in series(self, x, x0, n, dir, logx, cdir)
   2988 
   2989         if n is not None:  # nseries handling
-> 2990             s1 = self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
   2991             o = s1.getO() or S.Zero
   2992             if o:

~/sympy/sympy/core/power.py in _eval_nseries(self, x, n, logx, cdir)
   1623         maxpow = n - m*e
   1624 
-> 1625         if maxpow < S.Zero:
   1626             return O(x**(m*e), x)
   1627 

~/sympy/sympy/core/relational.py in __bool__(self)
    396 
    397     def __bool__(self):
--> 398         raise TypeError("cannot determine truth value of Relational")
    399 
    400     def _eval_as_set(self):

TypeError: cannot determine truth value of Relational

```
I will try to reproduce the issue with a simpler example and add it as a test.

Reply:
If there are no further suggested changes in this PR, think I will revert the relational operators -> assumptions changes and look into this in more detail, and open a new PR/close the issue according to what conclusion is drawn, because it is also unrelated to this PR.